libs/constrained-generators/src/Constrained/Examples/Map.hs

```diff
@@ -74,3 +75,18 @@ elemSpec = constrained' $ \k v m ->
       whenTrue (k' ==. k) (v' ==. v)
   , m `dependsOn` k
   ]
+
+lookupSpecific :: Specification BaseFn (Int, Int, Map Int Int)
+lookupSpecific = constrained' $ \k v m ->
+  [ m `dependsOn` k
+  , assert $ lookup_ k m ==. cJust_ v
+  ]
+
+specificElemConstraints :: Specification BaseFn Int
+specificElemConstraints = constrained $ \k ->
+  [ assert $ k `elem_` lit [1 .. 10]
+  , forAll' (lit $ Map.fromList [(i, i) | i <- [1 .. 10]]) $ \k' v ->
+      whenTrue
+        (k ==. k')
+        (v ==. 10)
+  ]
```

Comment:
This is meant for some kind of test?   Might explain what you expect this to do?

Reply:
I've removed it because it's not super interesting.